numpy/linalg/_linalg.py

```diff
@@ -197,6 +197,9 @@ def _assert_stacked_2d(*arrays):
 
 def _assert_stacked_square(*arrays):
     for a in arrays:
+        if a.ndim < 2:
```

Comment:
I like the combination. If one really wants to get out the most, it could be
```
try:
    m, n = a.shape[-2:]
except ValueError:
    riase LinAlgError(f"{a.ndim}-dimensional...") from None
if m != n:
   ...
```
Using that these days `try/except` has no cost if no exception is raised.

Reply:
Sorry, pushed this after submitting the review - above is the most relevant comment! (and not very relevant at that!)

Reply:
The `_assert_stacked_square` is about 10% faster using your suggestion, I updated the PR with this.